src/rgw/rgw_admin.cc

```diff
@@ -8166,52 +8166,37 @@ int main(int argc, const char **argv)
     if (opt_cmd == OPT::USER_LIST) {
       metadata_key = "user";
     }
-    void *handle;
-    int max = 1000;
-    int ret = store->meta_list_keys_init(dpp(), metadata_key, marker, &handle);
-    if (ret < 0) {
-      cerr << "ERROR: can't get key: " << cpp_strerror(-ret) << std::endl;
-      return -ret;
-    }
 
-    bool truncated;
+    bool truncated = false;
+    void *handle;
     uint64_t count = 0;
+    std::list<std::string> users;
 
-    if (max_entries_specified) {
-      formatter->open_object_section("result");
+    if (max_entries > 1000) {
+      max_entries = 1000;
     }
-    formatter->open_array_section("keys");
+    formatter->open_object_section("result");
 
-    uint64_t left;
-    do {
-      list<string> keys;
-      left = (max_entries_specified ? max_entries - count : max);
-      ret = store->meta_list_keys_next(dpp(), handle, left, keys, &truncated);
-      if (ret < 0 && ret != -ENOENT) {
-        cerr << "ERROR: lists_keys_next(): " << cpp_strerror(-ret) << std::endl;
-        return -ret;
-      } if (ret != -ENOENT) {
-	for (list<string>::iterator iter = keys.begin(); iter != keys.end(); ++iter) {
-	  formatter->dump_string("key", *iter);
-          ++count;
-	}
-	formatter->flush(cout);
+    formatter->open_array_section("keys");
+    int ret = store->list_user(dpp(), metadata_key, marker, max_entries, handle, &truncated, users);
+     if (ret < 0) {
+      cerr << "ERROR: can't get key: " << cpp_strerror(-ret) << std::endl;
+      return -ret;
+    }
+    if (ret != -ENOENT) {
+      for (std::list<std::string>::iterator iter = users.begin(); iter != users.end(); ++iter){
+        formatter->dump_string("key", *iter);
+        ++count;
```

Comment:
users.size() will give count in constant time as well.

Reply:
> `users.size()` will give count in constant time as well.

`count++` is constant operation thus not adding any time complexity to the overall loop.

Reply:
sorry, I am stretching a little, practically it might not have a smallest diff 😅 but we are incrementing the `count` variable `n` times in a loop, and `std::list<>` class is doing the same in the backend, so we are repeating it. 
`users.size()` will simply fetch that that value so will be O(1).

Reply:
Thats okay 😃 
I have removed `count++` and assigned count with `users.size()`
Also I have updated name of the function from `list_user()` to `list_users()`

Reply:
Great 👍, Thanks!!